critiquebrainz/frontend/external/musicbrainz_db/utils.py

```diff
@@ -0,0 +1,28 @@
+import critiquebrainz.frontend.external.musicbrainz_db.exceptions as mb_exceptions
+
+
+def get_entities_by_gids(query, entity_model, redirect_model, mbids):
+    """Get entities using their mbids.
+
+    Args:
+        query (Query): sqlalchemy Query object.
```

Comment:
The `query` may be modified for using statements like `joinedload` to make use of `join` statements and reducing the number of queries to the database.
Presently, due to SQLAlchemy's lazy loading, it makes an additional query for place type (https://github.com/metabrainz/critiquebrainz/blob/master/critiquebrainz/frontend/external/musicbrainz_db/serialize.py#L78) and another query for place area. (https://github.com/metabrainz/critiquebrainz/blob/master/critiquebrainz/frontend/external/musicbrainz_db/serialize.py#L87)